kcidb/test_oo.py

```diff
@@ -217,6 +217,25 @@ def query_str(self, pattern_string):
             kcidb.orm.Pattern.parse(pattern_string)
         )
 
+    @staticmethod
+    def filter_valid(container, type):
+        """
+        Filter out the valid items in a container.
+        Args:
+            container:  A dictionary whose values are object(s) that
+                        have "valid" property.
+            type:       A key in the container.
```

Comment:
Indexing a dictionary is a fundamental operation, and doesn't have anything to do with purely filtering a list. So, please, remove this argument, and explicitly retrieve the corresponding list from the container at the invocation point. Hiding such a trivial operation inside a function doesn't save us code, and makes it harder to understand. Let's make this function as simple as possible, while still being useful.

Reply:
Okay.